sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -860,6 +861,21 @@ pi_result piKernelGetSubGroupInfo(pi_kernel kernel, pi_device device,
   (void)param_value_size;
   size_t ret_val;
   cl_int ret_err;
+
+  std::shared_ptr<void> implicit_input_value;
+  if (param_name == PI_KERNEL_MAX_SUB_GROUP_SIZE && !input_value) {
+    // OpenCL needs an input value for PI_KERNEL_MAX_SUB_GROUP_SIZE so if no
+    // value is given we use the max work item sizes of the device to avoid
+    // truncation of max sub-group size.
+    implicit_input_value = std::shared_ptr<size_t[]>(new size_t[3]);
```

Comment:
Why heap-allocation instead of a plain (or std::) array on the stack?

Reply:
Generality, in case it is needed for similar situations in the future that isn't necessarily a `size_t[3]`. Honestly I am fine with either.